llvm/lib/CodeGen/GlobalISel/CombinerHelper.cpp

```diff
@@ -5122,6 +5149,29 @@ MachineInstr *CombinerHelper::buildUDivUsingMul(MachineInstr &MI) {
     return true;
   };
 
+  if (MI.getFlag(MachineInstr::MIFlag::IsExact)) {
+    // Collect all magic values from the build vector.
+    bool Matched = matchUnaryPredicate(MRI, RHS, BuildExactUDIVPattern);
+    (void)Matched;
+    assert(Matched && "Expected unary predicate match to succeed");
```

Comment:
I think it's a bit cleaner to write this type of assert as if (!match...) llvm_unreachable

Reply:
I am just mirroring the sdiv logic but changing what is necessary.

Reply:
Well both asserts would be written cleaner this way